### DISCUSSION TITLE
What is the best approach to Share kdbx with many different platforms in 2024?

### DISCUSSION THREAD

**Discussion by elRicharde** [keepassxreboot/keepassxc#10852] [Q&A]
Hi, i used KeePass2 and KeePass2Android which both are able to sync and work with one kdbx from different devices. 
Anyways i switched to KeePassXC on Linux and Windows, KeePassDX on Android and KeePassium on iOS - i like the benefits of these Apps and making me more OS-independent.
Until i figured out there is a different behavior in the programs i got a few Data-Losses. 
I read that @droidmonkey[ has no issues with using clouddrives](https://github.com/keepassxreboot/keepassxc/issues/2937#issuecomment-547222182) but i made different experiences... also found this thread [here on reddit saying it´s a problem with gdrive on mobile](https://www.reddit.com/r/KeePass/comments/11wvi9u/keepassdx_and_google_drive/)

So now, i tried syncthing, played around with KeeShare, made several tests... but i´m not very confident which is the approach to go. 
Like some users before me, i want to spread KeePassXC for many friends and family, but even if it is not easy - it needs to be fail safe at least - i mean db-updates shouldn´t be overwritten by accident. 

As I checked many discussions here, some tutorials there and also the q&a and Manual.. I miss the up to date best approach to synch 
- ios phone / tablet with 2 win pc
- android phone with 2 pc
- Linux pc with 2 Win pcs

I have storage in google drive, ms onedrive, Dropbox... just looking for an experienced user, an up to date guide and hope to find help here. Thanks a lot for your time, great work - please don´t see my post as a crtitic, i would love to share this great tools even more, but most useres have at least 2 devices and want to have there databases accessible from all devices. 

best regards
Richard

**Comment by droidmonkey** [comment]
What is the actual problem/concern? I didn't see it in your post besides a minor remark about overwrites.

**Reply by elRicharde** [reply]
Thanks for your quick reply. 
Using one Database for Android and PC, storing in gdrive got me overwrites. 
My Wife doesn´t want to learn about this even i tried hard, so now i manage her passwords and share them to her pc, via KeeShare and her apple phone.
My Concern is: KeeShare will not fix the issues using Gdrive i guess.
So i need to rethink how to share KeeFiles so i wonder if there is a proved and working concept in 2024 and which one might be the best. 
I´ve spent to much time already trying around myself, without being sure to be on the right way, so i ask here how people handle this and if they have issues. 
Thanks

**Reply by droidmonkey** [reply]
I use KeePassDX (Android) and sync to OneDrive without issue between my mobile and laptop. So the error is either with Google Drive, your android app, or a combo of the two. Generally speaking, we don't care what you use, but sync issues are absolutely not on our side.

**Reply by elRicharde** [reply]
I´m fine with that and i understand that you´re not the support for other apps. 
Just wanted to know which is the best approach for Other Users.
Do you use KeeShare or the Same DB in a oneDrive folder? 

If there is a User using more than 2 devices and also ios, windows, android combo i would love to hear the experience and setup. 

Thanks in advance and best regards
Richarde

**Comment by dheijl** [comment]
I'm also using KeepassDX on Android, and KeepassXC on Windows and Linux, and I do the syncing with Google Drive. I have not run in any problems yet. But I usually make the changes on Windows or Linux.

**Reply by elRicharde** [reply]
thanks for your response. 
Same question here: 
KeeShare is not implented in KeePassDX, so you can either use KeeShare to export to a shared DB which you need to open directly in your phone as there is no KeeShare or you just have one Database in your gdrive used by phone and pc.?

**Reply by dheijl** [reply]
I don't know about Keeshare, i just use the same kbdx directly from Google Drive, sorry.

**Reply by MadTooler** [reply]
> I don't know about Keeshare, i just use the same kbdx directly from Google Drive, sorry.

Curious, is your setup with or without the "Use alternate saving method" enabled?

I saw some losses last year between android keepassDX and keepassXC on windows with gdrive service.  I stopped using the android and am just now trying to look into if it is either something I was doing or a now resolved issue.  Where I was also synching between multiple windows PCs and gdrive, I did not see any losses.

**Reply by droidmonkey** [reply]
I have had problems with KeePassDX syncing the file with onedrive last year as well. I think it's all resolved now, haven't had an issue in the past 6 months.

**Reply by MadTooler** [reply]
Cool.  Good to here.

When would it be appropriate to use the "Use alternate saving method?"

I should mention, my data loss was after edits or new items from keepassXC directly.  I only viewed from DX.  It may have been related to differences in the "OK" and "Apply" buttons from within the entry editor.?

**Reply by droidmonkey** [reply]
Alternative saves would be used if you cannot use the normal method. With Google drive on windows you should have no problems. 

Both OK and Apply do the same thing, data is marked as modified and the database will be saved per your settings.
